target/linux/lantiq/dts/RS353.dts

```diff
@@ -0,0 +1,318 @@
+/dts-v1/;
+
+/include/ "vr9.dtsi"
+
+/ {
+	model = "RS353 - Bintec Elmeg RS353";
+
+	chosen {
+		bootargs = "console=ttyLTQ0,115200 init=/etc/preinit";
+		leds {
+                        running = &status;
+                        boot = &status;
+			wifi = &wlan;
+			usb2 = &usb; /* Front port is mapped to device 2 */
+			dsl = &dsl;
+                };
+
+	};
+
+	memory@0 {
+		reg = <0x0 0x8000000>;
+	};
+
+	sram@1F000000 {
+		cgu0: cgu@103000 {
+                        compatible = "lantiq,cgu-xway";
+                        reg = <0x103000 0x1000>;
+                        #clock-cells = <1>;
+			lantiq,phy-clk-src = <0x2>;
+		};
+	};
+
+	fpi@10000000 {
+		localbus@0 {
+			nor-boot@0 {
+				compatible = "lantiq,nor";
+				bank-width = <2>;
+				reg = <0 0x0 0x2000000>;
+				#address-cells = <1>;
+				#size-cells = <1>;
+
+				partitions {
+					compatible = "fixed-partitions";
+					#address-cells = <1>;
+					#size-cells = <1>;
+
+					boardconfig: partition@0 {
```

Comment:
> Du you prefer this option or the lzma-loader option?

No strong preference. But lzma would allow to you better compressed kernel images.

>  I can try it relatively easily if I know how to generate the kernel image.

It is only tested on LEDE but might work on OpenWrt as well:

```
define Build/lzma-loader
	rm -rf $(KDIR)/lzma-loader
	$(NO_TRACE_MAKE) -C lzma-loader \
		KDIR=$(KDIR) \
		LZMA_TEXT_START=0x2d6b0 \
		LOADADDR=0x80002000 \
		LOADER_DATA="$@" \
		compile
	cp $(KDIR)/lzma-loader/loader.bin $@
endef

define Device/RS353
  DEVICE_PROFILE := RS353
  DEVICE_TITLE := Bintec RS353
  KERNEL := kernel-bin | append-dtb | lzma
  IMAGE_SIZE := 31232k
  IMAGE/sysupgrade.bin = append-kernel | lzma-loader | append-rootfs | pad-rootfs
  DEVICE_PACKAGES := kmod-usb-dwc2 kmod-ath9k wpad-mini kmod-i2c-gpio kmod-swconfig wpad-mini kmod-ledtrig-usbdev
endef
TARGET_DEVICES += RS353
```

Run ```make target/linux/install V=s``` to see which commands are called.

> In a second time, I patched the bootmonitor to boot even if the CRC32 is not correct, and I generated an update image including the Patched bootmonitor and the u-boot + sysupgrade images.

Sound like something we shouldn't really do. Does the bootloader on the bintec provides commands to write to ram? Using u-boot if would suggest something like:

```
tftp $(loadaddr) openwrt.bin
erase 0xb00c0000 $(filesize)
cp.b $(loadaddr) 0xb00c0000 $(filesize)
```

The NOR flash is memory mapped to 0xb0000000.